lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -524,6 +564,36 @@ int amf_smf_send(
   return rc;
 }
 
+/***************************************************************************
+**                                                                        **
+** Name:    amf_validate_dnn()                                            **
+**                                                                        **
+** Description:                                                           **
+** This function validates the DNN string received from UE or from        **
+** mme.yml againt apn stored in amf_context for a particular imsi.        **
+***************************************************************************/
+int amf_validate_dnn(
+    const amf_context_s* amf_ctxt_p, std::string dnn_string, int* index,
+    bool ue_sent_dnn) {
+  // Validating apn_configuration_s
+  if (dnn_string.empty()) {
+    return RETURNok;
+  }
+  for (int i = 0;
+       i < (sizeof(amf_ctxt_p->apn_config_profile.apn_configuration) /
+            sizeof(amf_ctxt_p->apn_config_profile.apn_configuration[0]));
```

Comment:
why aren't you using the length information directly, i.e., `MAX_APN_PER_UE`?

Reply:
Sure, I'll update it.

Reply:
Comment addressed.